pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -281,44 +304,67 @@ func handleVolumesSnapshotStatusModify(ctx interface{}, key string, status inter
 		log.Errorf("Snapshot handling %s failed: %s", volumesSnapshotStatus.SnapshotID, volumesSnapshotStatus.Error)
 		appInstanceStatus.SetErrorWithSourceAndDescription(volumesSnapshotStatus.ErrorDescription, volumesSnapshotStatus.ErrorSourceType)
 		setSnapshotStatusError(appInstanceStatus, volumesSnapshotStatus.SnapshotID, volumesSnapshotStatus.ErrorDescription)
-		publishAppInstanceStatus(zedmanagerCtx, appInstanceStatus)
+		publishAppInstanceStatus(ctx, appInstanceStatus)
 		return
 	}
 	appInstanceStatus.SnapStatus.RollbackInProgress = false
-	publishAppInstanceStatus(zedmanagerCtx, appInstanceStatus)
+	// unpublish local app instance config, as the rollback is complete, we do not need to replace the config
+	config := lookupAppInstanceConfig(ctx, appInstanceStatus.Key(), false)
```

Comment:
If this returns nil it would be better to have a log.Error or even log.Fatal than the deference of nil below.

Reply:
Added a log.Fatal